### PR TITLE
movement-test: use dx/dy for diagonal movement

### PR DESCRIPTION
**Pull request by yberreby**

### CODE REVIEW

File: movement-test/src/main.c [hunk 1]
```diff
@@ -69,6 +69,8 @@ void main(void) {
     UINT8 frame_skip = 8;  // Update player's animation every 8 frame to
                            // slow down the animation (8 frames = ~133 ms
                            // between each animation frames)
+    UINT8 dx = 0;
+    UINT8 dy = 0;
```
Comment (yberreby) [inline movement-test/src/main.c:73]: It's debatable whether to make these global or not to save some stack space, but they should fit in registers...

Comment (ChatPion) [reply movement-test/src/main.c:73]: Other parts (like collisions!) will need these info ; they should be global (stored in a struct I think?)

File: movement-test/src/main.c [hunk 2]
```diff
@@ -97,32 +99,34 @@ void main(void) {
...
-            else if (player_direction == PLAYER_DIRECTION_LEFT) player_x -= 1;
-            else if (player_direction == PLAYER_DIRECTION_UP) player_y -= 1;
-            else if (player_direction == PLAYER_DIRECTION_DOWN) player_y += 1;
+
+            player_x += dx;
+            player_y += dy;
+
             move_sprite(PLAYER_SPRITE_ID, player_x, player_y);
 
             // We do not update the animation on each frame: the animation
```
Comment (ChatPion) [inline movement-test/src/main.c:132]: If the player is idle, the sprite should show the idle frame

File: movement-test/src/main.c (hunk 1 above)
Comment (yberreby) [reply movement-test/src/main.c:73]: Ah yes, I hadn't considered that the dx/dy might be needed elsewhere. You're right.

I don't know about the structs. I'm a bit wary of alignment...

File: movement-test/src/main.c (hunk 2 above)
Comment (yberreby) [reply movement-test/src/main.c:132]: That's already what it does.

Comment (ChatPion) [reply movement-test/src/main.c:132]: That's not the behaviour I see though

Comment (yberreby) [reply movement-test/src/main.c:132]: That's what I'm seeing under BGB though... Are you using VBAM?

Comment (yberreby) [reply movement-test/src/main.c:132]: https://webmshare.com/DBQ3Q

Comment (ChatPion) [reply movement-test/src/main.c:132]: Riiiight, there's no "idle" pose for lateral movements, my bad.

Comment (ChatPion) [reply movement-test/src/main.c:132]: I'm closing this thread and filing a new issue.

**Pull request merged**